google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1179,38 +1263,31 @@ public PooledSessionFuture get() {
   }
 
   class MultiplexedSessionFutureWrapper implements SessionFutureWrapper<MultiplexedSessionFuture> {
-    private SettableApiFuture<SessionReference> multiplexedSessionSettableApiFuture;
```

Comment:
We don't really need to save `SettableApiFuture<SessionReference>` per `MultiplexedSessionFutureWrapper`. This can directly be obtained from the global reference.